### ISSUE TITLE
Bug: Segmentation violation when running on Ubuntu 22.04

### ISSUE THREAD

**Issue by joelclimbsthings** [sammcj/gollama#45]
**Describe the bug**

I've tried using the pre-built release and building from source, but I receive the following error when running:

```
❯ ./gollama
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x545177]

goroutine 1 [running]:
net/url.(*URL).EscapedPath(0x0)
	/usr/local/go/src/net/url/url.go:705 +0x17
net/url.(*URL).JoinPath(0x0, {0xc00020dd58, 0x1, 0x8?})
	/usr/local/go/src/net/url/url.go:1201 +0x2f
github.com/ollama/ollama/api.(*Client).do(0xc00011c480, {0x8a3d68, 0xb7dc40}, {0x79cb06, 0x3}, {0x79ee2a?, 0x410305?}, {0x0?, 0x0?}, {0x721fa0, ...})
	/home/joelt/go/pkg/mod/github.com/ollama/ollama@v0.1.44/api/client.go:104 +0xe5
github.com/ollama/ollama/api.(*Client).List(0xc00011c480, {0x8a3d68, 0xb7dc40})
	/home/joelt/go/pkg/mod/github.com/ollama/ollama@v0.1.44/api/client.go:300 +0x85
main.main()
	/home/joelt/tools/gollama/main.go:112 +0x554

```

**To Reproduce**

Steps to reproduce the behaviour:

**Expected behaviour**

**Screenshots**

![image](https://github.com/sammcj/gollama/assets/444632/e18ac948-fe27-4206-b540-22cea70d8066)

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04
- Gollama 1.11.1

**Comment by sammcj** [comment]
Thanks for the bug report, I'll look into this!

@jralmaraz I wonder if it's related to recent logging/config changes, perhaps trying to access a log or config file that doesn't exist?

**Comment by sammcj** [comment]
@joelclimbsthings could you please try updating gollama (e.g. `go install github.com/sammcj/gollama@HEAD`) and let me know if the problem is resolved?

**Comment by joelclimbsthings** [comment]
@sammcj That worked! That's for the quick fix 🙇🏻 

I did get a small error regarding parsing the Ollama URL from the config, but fixed it be replacing `0.0.0.0:11434` with `http://0.0.0.0:11434`. And the linking worked, thanks!

**Comment by sammcj** [comment]
Ah great thanks for letting me know. I'll update the default config handler to make that a bit more reliable.
